spec/models/spree/order_spec.rb

```diff
@@ -673,11 +673,13 @@
 
     context "removing line_items" do
       it "updates shipping and transaction fees" do
-        # Setting quantity of an item to zero
+        # Set quantity of an item to zero
         order.update_attributes(line_items_attributes: [{id: order.line_items.first.id, quantity: 0}])
+        order.shipment.save
+        order.update_payment_fees!
+        order.update!
```

Comment:
I am not convinced order.update_attributes(line_items_attributes is a good way to change order attributes. But we must be using this somewhere in the code...

If we need this order.update_attributes(line_items_attributes, currently we have to do shipment.save, update_payment_fees and order.update! in the test (and probably later in the code) to fix the test (it's green), but it doesnt look like an acceptable solution.

What do you think?

Reply:
I'm wondering if `Spree::OrderUpdater` is what we want here. If not, then we probably want something similar to encapsulate the logic.

Reply:
I'm a bit confused too. What the test tells me is that updating just removing a line item through an `update_attributes` is not enough to update the shipping and transaction fees :thinking: